File: src/main/java/com/lonebytesoft/hamster/accounting/model/Category.java
package com.lonebytesoft.hamster.accounting.model;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;

@Entity
public class Category implements HasId<Long> {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    @Column
    private long id;

    @Column(unique = true)
    private String name;

    @Column
    private long ordering;

    @Override
    public Long getId() {
        return id;
    }

    public void setId(long id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public long getOrdering() {
        return ordering;
    }

    public void setOrdering(long order) {
        this.ordering = order;
    }

    @Override
    public String toString() {
        return "Category{" +
                "id=" + id +
                ", name='" + name + '\'' +
                ", order=" + ordering +
                '}';
    }

}

File: src/main/java/com/lonebytesoft/hamster/accounting/model/Category.kt
package com.lonebytesoft.hamster.accounting.model

import javax.persistence.Column
import javax.persistence.Entity
import javax.persistence.GeneratedValue
import javax.persistence.GenerationType
import javax.persistence.Id

@Entity
data class Category(

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    var id: Long = 0,

    @Column(unique = true, nullable = false)
    var name: String = "",

    @Column(nullable = false)
    override var ordering: Long = 0,

    @Column(nullable = false)
    var visible: Boolean = true

): Ordered
